net/net.go

```diff
@@ -293,6 +293,27 @@ func (n *net) AddThread(ctx context.Context, addr ma.Multiaddr, opts ...core.New
 		return
 	}
 
+	threadComp, err := ma.NewComponent(thread.Name, id.String())
+	if err != nil {
+		return
+	}
+	peerAddr := addr.Decapsulate(threadComp)
+	addri, err := peer.AddrInfoFromP2pAddr(peerAddr)
+	if err != nil {
+		return
+	}
+
+	// Check if we're trying to dial ourselves (regardless of addr)
+	addFromSelf := addri.ID.Pretty() == n.host.ID().Pretty()
```

Comment:
Here we're checking the ID, rather than the address. Since we can't dial ourself anyway, this seems like a broader check. Regardless of address, if we're trying to dial ourself, don't do it.

Reply:
Nit: I think you can avoid the `.Pretty()` calls, since `ID` is already a `string` (binary).